commit e1208547eba3d20630d1a116b670ef3751f3ef2f
Author: Orlando Vazquez <ovazquez@gmail.com>
Date:   2019-07-16T11:49:26-07:00 (3 months ago)
    
    TRITON-1749 sdc-migrate zfs quotas are perturbed by flexible disk

diff --git a/lib/backends/smartos/tasks/machine_migrate.js b/lib/backends/smartos/tasks/machine_migrate.js
index 28c27cd..c6b8dfc 100644
--- a/lib/backends/smartos/tasks/machine_migrate.js
+++ b/lib/backends/smartos/tasks/machine_migrate.js
@@ -293,23 +293,23 @@ function restoreZfsQuota(callback) {
 
     var vm = payload.vm;
 
-    // Determine what the current used size is, then set the zfs filesystem
-    // quota to that value.
+    // Determine what the current reservation size is, then set the zfs
+    // filesystem quota to that value.
     vasync.pipeline({arg: {}, funcs: [
-        function getUsedSize(ctx, next) {
+        function getReservationSize(ctx, next) {
             var cmd = '/usr/sbin/zfs';
             var args = [
                 'list',
                 '-Hp',
                 '-o',
-                'used',
+                'reservation',
                 vm.zfs_filesystem
             ];
 
             log.debug({cmd: cmd, args: args}, 'restoreZfsQuota');
 
             child_process.execFile(cmd, args, gExecFileDefaults,
-                    function _execZfsListUsedSizeCb(err, stdout, stderr) {
+            function _execZfsListReservationSizeCb(err, stdout, stderr) {
                 if (err) {
                     log.error('zfs list error:', err, ', stderr:', stderr);
                     next(new zfsError('zfs list failure', err, stderr));
